components/org.wso2.carbon.identity.data.publisher.application.authentication/src/main/java/org/wso2/carbon/identity/data/publisher/application/authentication/impl/DASLoginDataPublisherImpl.java

```diff
@@ -130,9 +130,15 @@ protected void publishAuthenticationData(AuthenticationData authenticationData)
 
         try {
             String roleList = null;
-            if (FrameworkConstants.LOCAL_IDP_NAME.equalsIgnoreCase(authenticationData.getIdentityProviderType())) {
-                roleList = getCommaSeparatedUserRoles(authenticationData.getUserStoreDomain() + "/" + authenticationData
-                        .getUsername(), authenticationData.getTenantDomain());
+            //User roleList retrieved as empty String when there is a previous authentication session available
+            // (different tab in the same browser). When there is a previous local authentication session,
+            //authenticationData.getIdentityProviderType() will be null.
+            if (FrameworkConstants.LOCAL_IDP_NAME.equalsIgnoreCase(authenticationData.getIdentityProviderType()) || (
```

Comment:
Can we refactor the complex if condition to private method "hasLocalIdPUsedForAuthentication(..)"

Reply:
fixed with https://github.com/wso2-extensions/identity-data-publisher-authentication/pull/73/commits/fbc9e667a0fdb9c0cf99dade8d797eb4c2d7d415